internal/pkg/gateway/commit/blocknotifier.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package commit
+
+import (
+	"sync"
+
+	"github.com/hyperledger/fabric/core/ledger"
+)
+
+type blockListener interface {
+	receiveBlock(*ledger.CommitNotification)
+	close()
+}
```

Comment:
Can we remove this interface and straight away use the two concrete types - to make it more explicit. The current code may have saved a few lines of code but lessen the readability.

Reply:
This abstraction isn't there to save a few lines of code. It's to:
1. Keep the blockListener [focused on the task](https://en.wikipedia.org/wiki/Single-responsibility_principle) of receiving and dispatching events from the ledger
2. Allow it to be [extended with other listeners without requiring code changes](https://en.wikipedia.org/wiki/Open%E2%80%93closed_principle)
3. Avoid it needing [dependencies on listener implementations](https://en.wikipedia.org/wiki/Dependency_inversion_principle)

Reply:
These objectives sounds good if they are between course level modules. Applying these to micro level of granularity has led to Fabric code at many places as hard to read IMO.

Reply:
Feel free to make that change